homeassistant/components/hue_ble/__init__.py

```diff
@@ -0,0 +1,59 @@
+"""Hue BLE integration."""
+
+import logging
+
+from HueBLE import HueBleLight
+
+from homeassistant.components.bluetooth import (
+    async_ble_device_from_address,
+    async_scanner_count,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the integration from a config entry."""
+
+    address = str(entry.data.get(CONF_MAC))
+
+    ble_device = async_ble_device_from_address(hass, address.upper(), connectable=True)
+
+    if ble_device is None:
+        count_scanners = async_scanner_count(hass, connectable=True)
+        _LOGGER.debug("Count of BLE scanners: %i", count_scanners)
+
+        if count_scanners < 1:
+            raise ConfigEntryNotReady(
+                "No Bluetooth scanners are available to search for the light."
+            )
+        raise ConfigEntryNotReady("The light was not found.")
```

Comment:
I saw this in one of the integrations I based this on and I kept it because I'd rather the error for not being able to find it and not being able to find it because there is nothing available to search for it be different things.

Reply:
I think this is somewhat valid, but I also think this kind of check should live in some BLE integration helper so not every integration has to duplicate such code.
@bdraco, do we have some helper like this already?